src/classes/RecipeRepository.js

```diff
@@ -1,8 +1,24 @@
+const data = require('../data/recipes')
+
 class RecipeRepository {
-  constructor() {
+  constructor(recipeData) {
+    this.recipeData = recipeData
 
     // One class to get you started!
   }
+
+  filterByTag(tag) {
+    let filteredRecipes = this.recipeData.filter(recipe => recipe.tags.includes(tag));
+    return filteredRecipes;
+  }
+
+  filterByName(name) {
+    let filteredRecipes = this.recipeData.filter(recipe => recipe.name === name);
+    return filteredRecipes;
+  }
+
+  // try reduce later for the extention using multiple parameters? Maybe use ternary operator
+
```

Comment:
.filter() is a great method to filter through recipeData. I agree with the .reduce() iterator method to filter "tags", "recipe names", "ingredients" to return a recipe; for the extension. Thank you for submitting. CG